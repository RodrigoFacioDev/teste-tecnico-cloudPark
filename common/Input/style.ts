import styled from "styled-components/native";

export const StyledInput = styled.TextInput`
  width: 100%;
  padding: 12px;
  border-width: 1px;
  border-color: #ccc;
  border-radius: 8px;
  margin-bottom: 16px;
  font-size: 16px;
`;
