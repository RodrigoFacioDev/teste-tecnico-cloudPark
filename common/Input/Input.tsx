import React from "react";
import { TextInputProps } from "react-native";
import { StyledInput } from "./style";

export default function Input(props: TextInputProps) {
  return <StyledInput placeholderTextColor="#999" {...props} />;
}
