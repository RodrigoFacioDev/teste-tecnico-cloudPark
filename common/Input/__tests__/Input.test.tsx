import { fireEvent, render } from "@testing-library/react-native";
import React from "react";
import Input from "../Input";

describe("<Input />", () => {
  it("should render the placeholder when provided", () => {
    const { getByPlaceholderText } = render(
      <Input placeholder="Type something..." value="" onChangeText={() => {}} />
    );
    expect(getByPlaceholderText("Type something...")).toBeTruthy();
  });

  it("should call onChangeText when the text changes", () => {
    const onChangeTextMock = jest.fn();
    const { getByPlaceholderText } = render(
      <Input placeholder="Test" value="" onChangeText={onChangeTextMock} />
    );
    fireEvent.changeText(getByPlaceholderText("Test"), "new text");
    expect(onChangeTextMock).toHaveBeenCalledWith("new text");
  });

  it("should match the component snapshot when rendered", () => {
    const tree = render(
      <Input placeholder="Snapshot" value="" onChangeText={() => {}} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
