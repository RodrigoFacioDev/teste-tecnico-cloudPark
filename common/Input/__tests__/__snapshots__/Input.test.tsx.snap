// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> faz snapshot do componente 1`] = `
<TextInput
  onChangeText={[Function]}
  placeholder="Snapshot"
  placeholderTextColor="#999"
  style={
    {
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderColor": "#ccc",
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "borderWidth": 1,
      "fontSize": 16,
      "marginBottom": 16,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "100%",
    }
  }
  value=""
/>
`;

exports[`<Input /> matches the component snapshot 1`] = `
<TextInput
  onChangeText={[Function]}
  placeholder="Snapshot"
  placeholderTextColor="#999"
  style={
    {
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderColor": "#ccc",
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "borderWidth": 1,
      "fontSize": 16,
      "marginBottom": 16,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "100%",
    }
  }
  value=""
/>
`;

exports[`<Input /> should match the component snapshot when rendered 1`] = `
<TextInput
  onChangeText={[Function]}
  placeholder="Snapshot"
  placeholderTextColor="#999"
  style={
    {
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderColor": "#ccc",
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "borderWidth": 1,
      "fontSize": 16,
      "marginBottom": 16,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "100%",
    }
  }
  value=""
/>
`;
